packages/jest-dom-mocks/src/match-media.ts

```diff
@@ -3,35 +3,32 @@ export interface MediaMatching {
 }
 
 export default class MatchMedia {
-  originalMatchMedia: (mediaQuery: string) => MediaQueryList;
-  private isUsingMockMatchMedia = false;
+  originalMatchMedia: ((mediaQuery: string) => MediaQueryList) | null = null;
 
   mock(media: MediaMatching = defaultMatcher) {
-    if (this.isUsingMockMatchMedia) {
+    if (this.originalMatchMedia !== null) {
```

Comment:
fyi we usually just use `== null` and `!= null` over their stricter versions since generally speaking an `undefined` is fine too, but I don't feel that strongly.